### PR TITLE
Removed usage of Instance

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/cli/ClientOpts.java [hunk 1]
```diff
@@ -19,13 +19,18 @@
...
 import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.client.impl.ClientContext;
 import org.apache.accumulo.server.client.HdfsZooInstance;
 
 public class ClientOpts extends org.apache.accumulo.core.cli.ClientOpts {
   {
     setPrincipal("root");
   }
 
+  public ClientContext getClientContext() {
```
Comment (ctubbsii) [inline server/base/src/main/java/org/apache/accumulo/server/cli/ClientOpts.java]: Not objecting to this, but what's this used for? Can it completely replace the other methods, like `getInstance()`?

File: server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java [hunk 2]
```diff
@@ -43,6 +43,7 @@
 import org.apache.accumulo.core.client.MutationsRejectedException;
 import org.apache.accumulo.core.client.Scanner;
 import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.impl.ClientContext;
```
Comment (ctubbsii) [inline server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java]: Should run a build without disabling the impsort plugin, just to make sure unused imports get removed.

File: server/gc/src/main/java/org/apache/accumulo/gc/replication/CloseWriteAheadLogReferences.java [hunk 3]
```diff
@@ -129,12 +129,10 @@ public void run() {
...
    * Construct the set of referenced WALs from zookeeper
    *
-   * @param conn
-   *          Connector
    * @return The Set of WALs that are referenced in the metadata table
    */
-  protected HashSet<String> getClosedLogs(Connector conn) {
-    WalStateManager wals = new WalStateManager(conn.getInstance(), ZooReaderWriter.getInstance());
+  protected HashSet<String> getClosedLogs() {
+    WalStateManager wals = new WalStateManager(context, ZooReaderWriter.getInstance());
```
Comment (ctubbsii) [inline server/gc/src/main/java/org/apache/accumulo/gc/replication/CloseWriteAheadLogReferences.java]: This looks right, but be careful here. Replication code has two "clients", and we need to make sure that if we're replacing any client API with an internal type, we're doing so with the "local" system, and not the client code which refers to the remote replica.

File: server/master/src/main/java/org/apache/accumulo/master/tableOps/CloneZookeeper.java [hunk 4]
```diff
@@ -32,9 +32,9 @@
 
   private CloneInfo cloneInfo;
 
-  public CloneZookeeper(CloneInfo cloneInfo) throws NamespaceNotFoundException {
+  public CloneZookeeper(CloneInfo cloneInfo, ClientContext context) throws NamespaceNotFoundException {
     this.cloneInfo = cloneInfo;
-    this.cloneInfo.namespaceId = Namespaces.getNamespaceId(HdfsZooInstance.getInstance(),
```
Comment (ctubbsii) [inline server/master/src/main/java/org/apache/accumulo/master/tableOps/CloneZookeeper.java]: +1000; I hate these.

File: core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java [hunk 5]
```diff
@@ -177,13 +177,18 @@ private static void getAllNamespaces(Instance instance, BiConsumer<String,String
...
-  public static boolean namespaceNameExists(Instance instance, String namespaceName) {
-    return lookupNamespaceId(instance, namespaceName) != null;
+  public static boolean namespaceNameExists(ClientContext context, String namespaceName) {
+    return lookupNamespaceId(context, namespaceName) != null;
   }
 
   /**
    * Look for namespace name in ZK. Throw NamespaceNotFoundException if not found.
    */
+  public static String getNamespaceName(ClientContext context, Namespace.ID namespaceId)
```
[the remark this answers is not in the document]
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java]: It's still needed but other methods

File: core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java [hunk 6]
```diff
@@ -145,8 +140,12 @@ public static String getTableOfflineMsg(Instance instance, Table.ID tableId) {
     }
   }
 
-  public static Map<String,Table.ID> getNameToIdMap(Instance instance) {
```
[the remark this answers is not in the document]
Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java]: It's still needed by other methods

File: server/base/src/main/java/org/apache/accumulo/server/cli/ClientOpts.java (hunk 1 above)
Comment (mikewalch) [reply server/base/src/main/java/org/apache/accumulo/server/cli/ClientOpts.java]: It's there to replace usages of `getInstance` but I haven't replaced them all yet.

File: server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java (hunk 2 above)
Comment (mikewalch) [reply server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java]: They got removed

File: server/gc/src/main/java/org/apache/accumulo/gc/replication/CloseWriteAheadLogReferences.java (hunk 3 above)
Comment (mikewalch) [reply server/gc/src/main/java/org/apache/accumulo/gc/replication/CloseWriteAheadLogReferences.java]: I checked and this code is OK. It's using the local system

**Pull request merged**